src/index.js

```diff
@@ -1,42 +1,80 @@
-    description: 'complete To Do List project',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'fix car',
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'go shopping',
-    completed: false,
-    index: 3,
-  },
-];
-
-const showTasks = () => {
-  const list = document.querySelector('.list');
+const input = document.getElementById('input');
+
+const target = document.getElementById('list');
+
+let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveToLocalStorage(data) {
+  tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.push(data);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+input.addEventListener('keyup', (event) => {
+  if (event.key === 'Enter') {
+    const inputValue = input.value;
+    const newObj = {
+      description: inputValue,
+      completed: false,
+      index: tasks.length + 1,
+    };
+    saveToLocalStorage(newObj);
+    window.location.reload();
+    input.value = '';
+  }
+});
+
+for (let i = 0; i < tasks.length; i += 1) {
+  const { index, description } = tasks[i];
+  target.innerHTML += `
+      <li id="L${index}" class ="common">
+      <input for ="P${index}" id="input" type="checkbox" class ="checkbox">
+      <p id ="P${index}" class="li-p">${description}</p>
+      <button id="edit-remove${index}"  class="btn dots list-item">
+       <i class="fa fa-ellipsis-v"></i>
+      </button>
+      </li>
+    `;
+}
+
+function deleteList() {
+  const button = document.querySelectorAll('.newButton');
+  const element = button[0].parentNode;
+  const liItem = element.parentNode;
+  liItem.removeChild(element);
+  const listId = element.getAttribute('id');
+  const index = parseInt(listId.substring(1), 10);
+  tasks = tasks.filter((task) => task.index !== index);
   for (let i = 0; i < tasks.length; i += 1) {
-    const listItems = document.createElement('div');
-    listItems.classList.add('to-do--');
-    listItems.innerHTML = `<div class="lists">
-    <input type="checkbox">
-    ${tasks[i].description} 
-  </div><div>
-    <i class="fa-solid fa-ellipsis-vertical"></i>
-    </div>
-  `;
-    list.appendChild(listItems);
+    tasks[i].index = i + 1;
   }
-};
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  window.location.reload();
+}
+
+function addTrash(event) {
+  const eventCatcher = event.target.closest('button');
+  eventCatcher.style.display = 'none';
+  const liItem = event.target.closest('li');
+  const nodeList = liItem.querySelectorAll('p');
+  const parentLi = nodeList[0].parentNode;
+  const parentId = parentLi.getAttribute('id');
+  const parentElement = document.getElementById(parentId);
+  const newBtn = document.createElement('button');
+  newBtn.className = 'newButton';
+  newBtn.innerHTML = '<i class="fa fa-trash-alt"></i>';
+  parentElement.append(newBtn);
+  liItem.style.backgroundColor = '#ffe6ee';
+  const textElem = liItem.querySelector('p');
+  textElem.setAttribute('contenteditable', true);
```

Comment:
- [ ] I am unable to edit a task. You only made the text editable but when I edit the description it is not persisted in the `localStorage`. Kindly fix that.